ci/test/00_setup_env_native_fuzz_with_msan.sh

```diff
@@ -18,7 +18,7 @@ export PACKAGES="ninja-build"
 export DEP_OPTS="DEBUG=1 NO_BDB=1 NO_QT=1 CC=clang CXX=clang++ CFLAGS='${MSAN_FLAGS}' CXXFLAGS='${MSAN_AND_LIBCXX_FLAGS}'"
 export GOAL="install"
 # _FORTIFY_SOURCE is not compatible with MSAN.
-export BITCOIN_CONFIG="--enable-fuzz --with-sanitizers=fuzzer,memory CPPFLAGS='-DBOOST_MULTI_INDEX_ENABLE_SAFE_MODE -U_FORTIFY_SOURCE'"
+export BITCOIN_CONFIG="-DFUZZ=ON -DSANITIZERS=fuzzer,memory -DAPPEND_CPPFLAGS='-DBOOST_MULTI_INDEX_ENABLE_SAFE_MODE -U_FORTIFY_SOURCE'"
```

Comment:
Are we using the `APPEND_*FLAGS` here just because setting `-DCMAKE_CXX_FLAGS` would clobber the `CXXFLAGS`?

Reply:
No.

`-DAPPEND_CPPFLAGS='-U_FORTIFY_SOURCE'` is required to override `-D_FORTIFY_SOURCE=3` from https://github.com/hebasto/bitcoin/blob/05784622feb15252a75dad64cdd5723e612e3f14/CMakeLists.txt#L505-L507

Reply:
FWIW, there was https://github.com/bitcoin/bitcoin/pull/30140.

Reply:
I guess the answer here is that given how hardening has been implemented in the CMake branch, it's impossible to override any of the flag using the proper CMake vars?

Reply:
> I guess the answer here is that given how hardening has been implemented in the CMake branch, it's impossible to override any of the flags using the proper CMake vars?

That's correct.

Reply:
That seems like a problem in it's own right, but I guess can be followed up with separately.

Reply:
Isn't this the exact use case we added the APPEND flags for, though?

Reply:
We'll probably just need to document which flags happen to only work with those flags in some way, because at the moment, it's buried in CMake implementation detail.

Reply:
On second glance, `-DBOOST_MULTI_INDEX_ENABLE_SAFE_MODE` is going into `DCMAKE_CXX_FLAGS` in most other places. If nothing else, we should be consistent with where we stick vars to avoid confusion.

Reply:
> On second glance, `-DBOOST_MULTI_INDEX_ENABLE_SAFE_MODE` is going into `DCMAKE_CXX_FLAGS` in most other places. If nothing else, we should be consistent with where we stick vars to avoid confusion.

Done.